src/js/components/user-management/signup.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link, Redirect } from 'react-router-dom';
+
+import { Button } from '@material-ui/core';
+
+import ReCAPTCHA from 'react-google-recaptcha';
+
+import { setSnackbar } from '../../actions/appActions';
+import { createOrganizationTrial } from '../../actions/organizationActions';
+import { loginUser } from '../../actions/userActions';
+
+import Form from '../common/forms/form';
+import TextInput from '../common/forms/textinput';
+import PasswordInput from '../common/forms/passwordinput';
+import FormCheckbox from '../common/forms/formcheckbox';
+import { WelcomeSnackTip } from '../helptips/onboardingtips';
+import { getOnboardingStepCompleted } from '../../utils/onboardingmanager';
+
+export class Signup extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      step: props.step || 1,
+      email: '',
+      password: '',
+      password_confirmation: '',
+      name: '',
+      tos: '',
+      marketing: '',
+      recaptcha: '',
+    };
+  }
+
+  _handleStep1(formData) {
+    if (formData.password_new != formData.password_confirmation) {
+      this.props.setSnackbar('The passwords you provided do not match, please check again.', 5000, '');
```

Comment:
sidenote: this would preferably be handled by the form but this should be a lot easier once https://tracker.mender.io/browse/MEN-3530 is done - so please don't _disregard after reading_ 🙂

Reply:
I desperately tried to find a way to get complex validations work, but then I gave up and implemented it in this way. If you have a suggestion on how to implement such validation logic, I'm all ears.

Reply:
unfortunately I'm not aware of a "pretty" path with the current setup, it might be possible to check this in the `validateForm`, but only if the passwords were tied to an additional info to mark the dual components

Reply:
ok, then let's skip this thing and eventually handle it later on when refactoring of forms is over